File: main (2).swift
import Foundation

func reverseNumber(_ num: Int) -> Int {
    return Int(String(String(num).reversed()))!
}

let n = Int(readLine()!)!
var prev = Int(readLine()!)!
var result: [Int] = []

for _ in 1..<n {
    let current = Int(readLine()!)!
    if current < prev {
        result.insert(reverseNumber(prev), at: 0) // Добавляем в начало
    }
    prev = current
}

result.insert(reverseNumber(prev), at: 0) // Добавляем последнее число

// Переворачиваем и выводим
print(result.reversed().map { String($0) }.joined(separator: " "))

File: main (2).cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int reverseNumber(int num) {
    int reversed = 0;
    while (num > 0) {
        reversed = reversed * 10 + num % 10;
        num /= 10;
    }
    return reversed;
}

int main() {
    int n;
    cin >> n;

    int prev;
    cin >> prev;
    vector<int> result;

    for (int i = 1; i < n; i++) {
        int current;
        cin >> current;
        if (current < prev) {
            result.insert(result.begin(), reverseNumber(prev)); // Добавляем в начало
        }
        prev = current;
    }

    result.insert(result.begin(), reverseNumber(prev)); // Добавляем последнее число

    for (auto it = result.rbegin(); it != result.rend(); ++it) {
        cout << *it << " ";
    }
    cout << endl;
}
